app/controllers/posts_controller.rb

```diff
@@ -1,7 +1,8 @@
 class PostsController < ApplicationController
   def index
     @user = User.find(params[:user_id])
-    @posts = @user.posts
+    # Eager loading comments and likes for the user's posts
+    @posts = @user.posts.includes(:comments, :likes)
```

Comment:
- [ ] @Team great work fixing N +1 problem, however, you forgot to add screenshots of your app's logs before and after the fix in the pull request description, kindly consider implementing this, its part of the project requirement as shown below
![image](https://github.com/serengia/rails-blog-app/assets/61408860/f655b552-9ecd-4d44-8af1-04a17f9288cc)